testng-core-api/src/main/java/org/testng/IObjectFactory2.java

```diff
@@ -17,7 +19,7 @@ public interface IObjectFactory2 extends ITestObjectFactory {
    * @return - The newly created object.
    */
   @Deprecated
-  default Object newInstance(Class<?> cls) {
+  default Object newInstance(@NonNull Class<?> cls) {
```

Comment:
It would be better to use package-level "default not null" annotation rather than `@NotNull`: https://github.com/pgjdbc/pgjdbc/blob/aff581fe806f1c1cdc8bcd9e5a981e37691b44b5/pgjdbc/src/main/java/org/postgresql/package-info.java#L6-L8

See explanation in https://checkerframework.org/manual/#climb-to-top